### PR TITLE
Enhancing Quote block styling for different text alignments

### LINKED ISSUE #11609: Right-align border for right-aligned quotes
To improve the consistency of alignments and to avoid creating potentially-awkward rivers inside the quote block, I'd recommend adjusting the positioning of the hard left border on the quote block when the alignment is changed, so it would be to the right on a right-aligned quote, and either removed or maybe at the top of a centre-aligned block.

**Left-aligned**

screenshot 2018-11-08 01 07 40

Suggestion is to remain as-is, since the alignment here is consistent with the text.

**Centre-aligned**

screenshot 2018-11-08 01 07 29

Suggestion:
screenshot 2018-11-08 01 08 57

**Right-aligned**
screenshot 2018-11-08 01 07 50

Suggestion:
screenshot 2018-11-08 01 08 17

### PR DESCRIPTION
**Pull request by colorful-tones**
Fixes #11609

## Description
These changes are CSS only, and in the Quote block's `theme.scss`, which is loaded in both the editor _and_ on the front end if theme has opted in to using them. See Gutenberg Handbook: [Theme Support](https://wordpress.org/gutenberg/handbook/designers-developers/developers/themes/theme-support/#default-block-styles).

The CSS affects and/or adds a border to either the left or right depending on the `text-align` chosen in the editor. Border on `text-align: center` are removed.

## How has this been tested?
Changes were tested in local Docker WP environment with Gutenberg post with all variations of Quote block added. Changes were tested against the TwentyFifteen, TwentySixteen, TwentySeventeen and TwentyNineteen themes.

Tested in Chrome 71.0.3578.98 on macOS 10.14.1.

## Screenshots

twentyfifteen-quote-block-gutenberg-style
twentyfifteen-quote-block-style
twentynineteen-quote-block-gutenberg-style
twentynineteen-quote-block-style
twentyseventeen-quote-block-gutenberg-style
twentyseventeen-quote-block-style
twentysixteen-quote-block-gutenberg-style
twentysixteen-quote-block-style

## Types of changes
* New feature (non-breaking change which adds functionality): adds a right border if Quote block is `text-align: right`. However, this does have some impact on existing editor and/or theme styles for _some_ TwentySomething themes:
    1. TwentyFifteen, TwentySixteen and TwentyNineteen: the `editor-blocks.css` (Gutenberg editor styles) may need to be updated (see screenshot above). The front end output is fine for TwentyFifteen and TwentySixteen.
    2. TwentyNineteen: the theme's styling will need to be updated, because it will now show both:
        * `border-left: 2px solid #0073aa`
        * `border-right: 4px solid #000;

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by jasmussen** [comment]
THANK YOU for sticking with this! This seems overall like a very solid pull request, I'm sure we'll get this in relatively quickly. 🏅

This seems to be working on the frontend, but I'm not seeing any changes in the editor. Have you tried a theme that does not provide editor styles and relies on the default vanilla editor stylesheet? I'm not sure why the theme.scss files aren't being picked up. @gziolo any idea?

changes

Styling using this selector, `[style="text-align:right"]` seems a little funky to me. I mean if it works, and is compatible, it's probably fine. But let's sanity check this with some of the others... @kjellr any insights?

For example for RTL languages, we want to make sure that _explicitly set alignments_ are not autoreversed by PostCSS. For example here's what we have for floats:

```
	.alignleft {
		/*rtl:ignore*/
		float: left;
		/*rtl:ignore*/
		margin-right: 1em;
	}

	.alignright {
		/*rtl:ignore*/
		float: right;
		/*rtl:ignore*/
		margin-left: 1em;
	}
```

The `/*rtl:ignore*/` comment ensures that the line right after is ignored by the RTL engine that would otherwise automatically rewrite `float: left` to `float: right` in the `*-rtl.css` file. We want these to _not be rewritten_ because when it's an explicit choice by the user to align something right, it doesn't matter which direction it was initially.

Which brings us to the next thing we need to sanity check and be sure of. The blockquote comes with left alignment by default, which is later overrwritten (`border-left: none` and `border-right: something` in the right alignment choice). Which means those defaults _should_ be rewritten by the postCSS RTL engine. But we want to make sure whatever is rewritten here, is correctly overwritten when the user makes the aforementioned explicit choice.

In other words, I think what we need to do here is to:

- Get this to work in the editor
- Be sure to test every possible alignment, including _no_ alignment, in LTR and RTL text directions both.

Nice work! ✨

**Comment by kjellr** [comment]
> Styling using this selector, [style="text-align:right"] seems a little funky to me. I mean if it works, and is compatible, it's probably fine. But let's sanity check this with some of the others... @kjellr any insights?

Yeah, I remember running into that recently, too. As far as I know, that's the only way to style these sorts of text alignments within the editor right now — Gutenberg don't provide a special class to hook into for alignment.  

As for other notes, I'm in agreement with Joen: This seems solid as long as we can make sure it's working great in the editor (not working for me there either yet), for RTL.

It's worth highlighting again that this may be a small breaking change for some themes. For instance, Twenty Nineteen will need a patch to make sure people don't end up with double borders as noted above.

**Comment by jasmussen** [comment]
> It's worth highlighting again that this may be a small breaking change for some themes. For instance, Twenty Nineteen will need a patch to make sure people don't end up with double borders as noted above.

That's a good point. Is there any way we can achieve the same effect, without having to break things here?

**Comment by colorful-tones** [comment]
Just wanted to drop a status: I’m off on vacation this entire week. I’ll be back next week and will see if I can schedule this in. I’ll drop another update early next week (Jan. 28). Thanks!

**Comment by jasmussen** [comment]
No stress, and thanks for your work!

**Comment by gziolo** [comment]
@jasmussen I see some of the styles used in the editor but overridden:


screen shot 2019-01-23 at 18 56 41

**Comment by jasmussen** [comment]
Indeed, good catch. It seems the editor style needs looser specificity, if possible, to let the theme style more easily override.

**Comment by colorful-tones** [comment]
Looks like it is taking me a bit longer to circle back on this. I'm hoping in next week or two to get an update on the Pull Request. Thanks for all the testing and feedback!

**Comment by m-e-h** [comment]
Great idea!!

The reason it's not working is because attribute selectors are kinda funny.

The space between `text-align:` and `right` matter. Which is how it's done in the editor HTML.
The front end doesn't have the space, so what you have here should work there since it's an exact match.

You'll probably also need a general `*` regex thingy when working with the space.

Try this: `.wp-block-quote[style*="text-align: right"]` for the editor style.

**Comment by m-e-h** [comment]
Sorry @colorful-tones, I probably didn't explain very well, but I think you're still going to need what you had previously to target the front end styles. The space is to target the editor. I don't think the space appears in the front-end. I'm guessing you'll need both:
```
&[style="text-align:right"],
&[style*="text-align: right"] {
```

```
&[style="text-align:center"],
&[style*="text-align: center"] {
```
FYI, I haven't tested this.

**Comment by colorful-tones** [comment]
I updated the selector in `theme.scss` from this:

```
	&[style="text-align:right"] {
		border-left: none;
		border-right: 4px solid $black;
		padding-left: 0;
		padding-right: 1em;
	}
```

To this:

```
	&[style*="text-align:right"],
        &[style*="text-align: right"] {
		border-left: none;
		border-right: 4px solid $black;
		padding-left: 0;
		padding-right: 1em;
	}
```

Kudos to @m-e-h for pointing that out! 

As far as what the LTR/RTL expectations are... I'm still unclear. 

I've tested with **TwentySixteen in editor** and when switching to RTL the following occurs:
* Default state - Text is aligned right, which is to be expected. No borders and a negative, right margin (from theme's `twentysixteen/css/editor-blocks.css`)
* Align Text: Right - Has the same exact appearance as the default state: text is aligned right, no borders and a negative, right margin.
* Align Text: Left - Has the same exact appearance as the default state (no borders and negative, right margin), but text is aligned to the left.
* Align Text: Center - Again, has same exact appearance as default state (no borders and negative, right margin), but text is aligned to the center.

I'll drop back and continue testing against TwentySomething themes, as well as a few others from WP.org repo.

It is very likely this change will impact themes and there is quite little we can do to minimize impact.

**Comment by jasmussen** [comment]
Thank you @m-e-h and @colorful-tones! I think this is good to go, styles look great in the editor.

LTR:

quote block

RTL:

rtl

Themes like TwentyNineteen need to update themselves here, but this is probably okay. 

I could use a final sanity check from @kjellr, what do you think?

File: packages/block-library/src/quote/theme.scss [hunk 1]
```diff
@@ -10,9 +12,21 @@
...
-	padding-left: 1em;
+	&[style*="text-align:right"],
+	&[style*="text-align: right"] {
+		border-left: none;
+		border-right: 4px solid $black;
+		padding-left: 0;
+		padding-right: 1em;
+	}
+
+	&[style="text-align:center"] {
```
Comment (kjellr) [inline packages/block-library/src/quote/theme.scss]: I think this one also needs to include the alternate selector with a space added: `&[style*="text-align: center"]`. I'm not seeing the border disappear otherwise. 

After that's updated, this should be all set.

**Comment by kjellr** [comment]
Just needs one minor update, and then this is good to go. I'm happy to submit a patch to Twenty Nineteen to get it working as expected after this change is in place. 👍

**Comment by colorful-tones** [comment]
@kjellr I've updated the `text-align: center` attribute selector. 

Keep in mind that this does not impact TwentyNineteen, because it is targeting:

```
.editor-styles-wrapper .wp-block-quote:not(.is-large):not(.is-style-large) {
    border-left: 2px solid #0073aa;
}
```

Therefore, TwentyNineteen's `text-align: center` appearance stays true.

screenshot 2019-02-12 09 45 56

**Comment by kjellr** [comment]
> Therefore, TwentyNineteen's `text-align: center` appearance stays true.

Yep! Makes sense. The patch I submitted for Twenty Nineteen just fixes the double borders. Everything else works as expected: 

Before the Twenty Nineteen Patch: 
screen shot 2019-02-12 at 10 03 01 am

After the Twenty Nineteen Patch: 
screen shot 2019-02-12 at 10 02 08 am

**Review by kjellr — APPROVED** [review]
I gave this another round of testing and it looks good to me. Thank you, @colorful-tones! 🚢

**Comment by colorful-tones** [comment]
Thanks @kjellr for following through with TwentyNineteen patch. 🥇

**Comment by m-e-h** [comment]
Really liking the added polish this feature gives the quote block. Great idea @colorful-tones !

**Comment by sarahmonster** [comment]
Thanks so much for sticking this one though, @colorful-tones! It's great to see this implemented. 👏
